classes/Product.php

```diff
@@ -273,7 +274,7 @@ class ProductCore extends ObjectModel
      *          - 1 Allow orders
      *          - 2 Use global setting
      */
-    public $out_of_stock;
+    public $out_of_stock = OutOfStockType::OUT_OF_STOCK_DEFAULT;
```

Comment:
doesn't it count as a BC break?

Reply:
No because this is the default value that is set in the `add` function anyway So I just emphasized its default value in the class And I added additional behat tests to ensure these default values at creation